packages/backend/src/services/ProjectService/ProjectService.ts

```diff
@@ -2621,4 +2622,50 @@ export class ProjectService {
             },
         );
     }
+
+    async calculateTotal(
+        user: SessionUser,
+        projectUuid: string,
+        data: any,
+    ): Promise<Record<string, any>> {
+        const { organizationUuid } = await this.projectModel.getSummary(
+            projectUuid,
+        );
+        if (
+            user.ability.cannot(
+                'view',
+                subject('Project', { organizationUuid, projectUuid }),
+            )
+        ) {
+            throw new ForbiddenError();
+        }
+
+        const baseQuery: MetricQuery = data.metricQuery;
+        const totalQuery: MetricQuery = {
+            ...baseQuery,
+            tableCalculations: [],
+            sorts: [],
+            dimensions: [],
+            metrics: baseQuery.metrics,
+            additionalMetrics: baseQuery.additionalMetrics,
+        };
+
+        const exploreName = data.explore;
+        const sql = await this.compileQuery(
+            user,
+            totalQuery,
+            projectUuid,
+            exploreName,
+        );
+
+        const results = await this.runMetricQuery({
+            user,
+            projectUuid,
+            metricQuery: totalQuery,
+            exploreName,
+            csvLimit: undefined,
+            context: QueryExecutionContext.CALCULATE_TOTALS,
+        });
+        return { results: results.rows[0], sql: sql.query };
```

Comment:
So @TuringLovesDeathMetal can debug this, it will be removed before merging https://lightdash.slack.com/archives/C065Y8BPMHA/p1700125428815639?thread_ts=1700122642.358299&cid=C065Y8BPMHA